src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -377,13 +377,30 @@ def cleanRSE(self, rse):
                             self.logger.exception(msg, rse['name'], str(ex))
 
                     self.logger.info("RSE: %s, Dir: %s, filesDeletedSuccess: %s",
-                                      rse['name'], dirLfn, deletedSuccess)
-                    rse['counters']['filesDeletedSuccess'] += deletedSuccess
-
-                    # Now clean the whole branch
-                    self.logger.debug("Purging dirEntry: %s:\n", dirPfn)
-                    purgeSuccess = self._purgeTree(ctx, dirPfn)
-                    if purgeSuccess:
+                                      rse['name'], dirLfn, filesDeletedSuccess)
+                    rse['counters']['filesDeletedSuccess'] += filesDeletedSuccess
+
+                    # Now delete the whole branch
+                    # First try to delete the base directory:
+                    rmdirSuccess = False
+                    purgeSuccess = False
+                    try:
+                        self.logger.debug("Trying to remove Dir: %s", dirPfn)
+                        # NOTE: For gfal2 rmdir() exit status of 0 is success
+                        rmdirSuccess = ctx.rmdir(dirPfn) == 0
+                    except gfal2.GError as gfalExc:
+                        if gfalExc.code == errno.ENOENT:
```

Comment:
can you please clarify and/or add a pointer to what `ENOENT` error is?

Reply:
https://man7.org/linux/man-pages/man3/errno.3.html :
`ENOENT No such file or directory (POSIX.1-2001).`